baseplate/events/queue.py

```diff
@@ -0,0 +1,146 @@
+"""Client library for sending events to the data processing system.
+
+This is for use with the event collector system. Events generally track
+something that happens in production that we want to instrument for planning
+and analytical purposes.
+
+Events are serialized and put onto a message queue on the same server. These
+serialized events are then consumed and published to the remote event collector
+by a separate daemon.
```

Comment:
In regular operation? no more so than we already do (`amqp.add` adds an item to a `Queue.Queue` in-process which gets drained by the worker thread that talks to amqp synchronously).

When things are going poorly? yes, that's definitely a concern. We can pause the scaler during an extended downtime of the collector.